File: Som/Network/INeuron.cs
using System;
using System.Collections.Generic;
using Som.ActivationFunction;

namespace Som.Network
{
    public interface INeuron
    {
        double[] Weights { get; set; }

        double GetReaction(double[] inputVector);

        IActivationFunction ActivationFunction { get; set; }
    }
}

File: JavaSom/Network/INeuron.java
package Network;

import ActivationFunction.*;

	public interface INeuron
	{
		double[] getWeights();
		void setWeights(double[] value);

		double GetReaction(java.util.List<Double> inputVector);

		IActivationFunction getActivationFunction();
		void setActivationFunction(IActivationFunction value);
	}